code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm

```diff
@@ -322,158 +323,158 @@
 				if(MOONSHINE)
 					icon_state = "glass_clear"
 					item_state = "glass_clear"
-					name = "\improper Moonshine"
+					name = "moonshine"
 					desc = "You've really hit rock bottom now... your liver packed its bags and left last night."
 				if(SOYMILK)
 					icon_state = "glass_white"
 					item_state = "glass_white"
 					name = "glass of soy milk"
 					desc = "White and nutritious soy goodness!"
 				if(BERRYJUICE)
-					icon_state = BERRYJUICE
-					item_state = BERRYJUICE
+					icon_state = "berryjuice"
+					item_state = "berryjuice"
```

Comment:
why are you getting rid of all the defines?

Reply:
I explain this in the OP. Using the defines for icon_state and item_state means that if the define was to be changed for whatever reason, it would break the item's icon unless the .dmi was also changed. It's better to refer to the icon by the name it is in the .dmi.

Reply:
to be clear, using defines to fetch a reagent-specific state is *fine*
but why bother when we're not using it for the obvious benefit of automatically getting item state without having five billion switch cases, e.g.:
```
icon_state = reagents.get_master_reagent_id()
item_state = reagents.get_master_reagent_id()
```
which is two lines instead of almost 600